### PR TITLE
Run unittests with GitHub actions

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

**Comment by hardbyte** [comment]
I can't believe they ran on the first attempt. There must be some trickery going on.

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        shell: bash
+        run: |
+          python -m pip install -U -r requirements.txt
+      - name: Build and install anonlink
+        shell: bash
+        run: |
+          python -m pip install -e .
+      - name: Test with pytest
+        run: |
+          pytest -q
```
Comment (wilko77) [inline .github/workflows/unittests.yml:39]: The only extra thing of value that Travis did for us is running the extended test. Maybe we should add that here.
```suggestion
          pytest -q
      - name: Test with extended size 100k
        if: ${{ matrix.python-version == 3.9}}
        env: 
          INCLUDE_100K: 1
        run: |
          pytest -q
```

**Review by wilko77 — COMMENTED** [review]
very bare bones.
If we want to replace azure pipelines, we also need to include codecov. 
Also, previously we would build the wheels first and then test with the wheels. Do we want to do that here as well?

For just replacing Travis there are 2 things:
- the extended test (see my code suggestion)
- test on pypy3, which was allowed to fail.
Don't know if Actions support pypy3. Also, there used to be problems with defining actions that are allowed to fail. Not sure if they support that yet.

I'm happy with not testing pypy3 for now.

**Comment by hardbyte** [comment]
> very bare bones.
> If we want to replace azure pipelines, we also need to include codecov.

I wasn't aware we were replacing azure pipelines, it was just travis-ci.org was no longer running our extended tests. I've added other operating systems though so we could chip away at azure's responsibilities.

> Also, previously we would build the wheels first and then test with the wheels. Do we want to do that here as well?
> 

`pip install -e .` should be fine for testing the code. Agree that we still need tests  for the wheel itself before publishing to pypi.

>  ...
> I'm happy with not testing pypy3 for now.

Yeah I tried it but it failed and it seems that Github actions indeed don't deal well with allowed failures.

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python Test and Lint
```
Comment (wilko77) [inline .github/workflows/unittests.yml:4]: we don't do linting, or do we?

File: .github/workflows/unittests.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          python -m pip install -U -r requirements.txt
+      - name: Build and install anonlink
+        shell: bash
+        run: |
+          python -m pip install -e .
+      - name: Test with pytest
+        run: |
+          pytest -q
+      - name: Test with extended size 100k
+        if: ${{ matrix.python == 3.9}}
```
Comment (wilko77) [inline .github/workflows/unittests.yml:43]: now that we have quite a few os-es, should we restrict that further? Do we really need to run a 5 minute test on every platform?

**Review by wilko77 — APPROVED** [review]

File: .github/workflows/unittests.yml (hunk 2 above)
Comment (hardbyte) [reply .github/workflows/unittests.yml:4]: Good point, I'll rename. I tried running black and flake8 but passing either would require lots of code changes

**Pull request merged**